app/src/main/java/com/lastaoutdoor/lasta/ui/screen/login/components/LoginContent.kt

```diff
@@ -73,3 +82,144 @@ fun LoginContent(onLoginClick: () -> Unit) {
+        // pages of content
+        HorizontalPager(
+            state = pagerState,
+            modifier = Modifier.testTag("HorizontalPager").fillMaxHeight(0.9f)) { page ->
+              Column(
+                  modifier = Modifier.fillMaxWidth().fillMaxHeight(),
+                  horizontalAlignment = Alignment.CenterHorizontally,
+                  verticalArrangement = Arrangement.SpaceEvenly) {
+                    PagerContent(page = page)
+                  }
+            }
+        HorizontalPagerIndicator(
+            pageCount = pagerState.pageCount,
+            currentPage = pagerState.currentPage,
+            targetPage = pagerState.targetPage,
+            currentPageOffsetFraction = pagerState.currentPageOffsetFraction)
+      }
+}
+
+@Composable
+private fun HorizontalPagerIndicator(
+    pageCount: Int,
+    currentPage: Int,
+    targetPage: Int,
+    currentPageOffsetFraction: Float,
+    unselectedIndicatorSize: Dp = 12.dp,
+    selectedIndicatorSize: Dp = 14.dp,
+    indicatorPadding: Dp = 6.dp
+) {
+  Row(
+      horizontalArrangement = Arrangement.Center,
+      verticalAlignment = Alignment.CenterVertically,
+      modifier =
+          Modifier.wrapContentSize()
+              .height(selectedIndicatorSize + indicatorPadding * 2)
+              .testTag("pageIndicator")) {
+        val indicatorColor = MaterialTheme.colorScheme.primary
+
+        // draw an indicator for each page
+        repeat(pageCount) { page ->
+          // calculate color and size of the indicator
+          val (color, size) =
+              if (currentPage == page || targetPage == page) {
+                // calculate page offset
+                val pageOffset = ((currentPage - page) + currentPageOffsetFraction).absoluteValue
+                // calculate offset percentage between 0.0 and 1.0
+                val offsetPercentage = 1f - pageOffset.coerceIn(0f, 1f)
+
+                val size =
+                    unselectedIndicatorSize +
+                        ((selectedIndicatorSize - unselectedIndicatorSize) * offsetPercentage)
+
+                indicatorColor.copy(alpha = offsetPercentage) to size
+              } else {
+                indicatorColor.copy(alpha = 0.1f) to unselectedIndicatorSize
+              }
+
+          Canvas(
+              modifier =
+                  Modifier.padding(
+                      // apply horizontal padding, so that each indicator is same width
+                      horizontal =
+                          ((selectedIndicatorSize + indicatorPadding * 2) - size) / 2 + 8.dp,
+                      vertical = size / 4),
+              onDraw = { drawCircle(color = color, radius = size.toPx() / 2) })
+        }
+      }
+}
+
+// Content that will be displayed in the pager pages
+@Composable
+fun PagerContent(page: Int) {
+  when (page) {
+    0 -> {
+      // Fist page : LOGO + ask to swipe
+      Image(
```

Comment:
Here you could write a subfunction to produce each page, e.g. `LogoDisplayPage()`, `ActivityTutoPage()`, etc

Reply:
You are right, I created functions for reach page